earthpy/spatial.py

```diff
@@ -30,14 +31,15 @@ def extent_to_json(ext_obj):
     """
 
     if type(ext_obj) == gpd.geodataframe.GeoDataFrame:
-        extent_json = mapping(box(*ext_obj.bounds.values[0]))
+        extent_json = mapping(box(*ext_obj.total_bounds))
```

Comment:
I think this `total_bounds` attribute is more general than `bounds.values[0]`, because it will work with points (GeoDataFrames with multiple rows) and polygons alike, but maybe you should make sure this still works for your intended use cases @lwasser

Reply:
oh you're right @mbjoseph that value makes no sense for a single point. (bounds) then again this is a note for me as what happens if this is fed a shapefile with only one point?!!! i'm not sure i thought of that edge case at all.

Reply:
agreed!!

missing edge case: if a user provides a single point geodataframe... what happens? It should fail gracefully. Should we provide it that in a test of some sort? i totally missed that edge case.

Reply:
Good question! If a user gives a single point GeoDataFrame, then they get a bounding "point" - basically a bounding box where the lower left and upper right corners are on the point. I would lean towards not raising an exception for this case, to keep this function as general as the geopandas backend (another way of saying that if geopandas allows it, we might as well allow it too).

Reply:
One other bit of weirdness that I will fix though is that `shapely.geometry.box` does not raise exceptions when minx is greater than maxx, or when miny is greater than maxy. For that, I think it would make sense to add an exception to help users out in case they mix up x and y.

Reply:
Right but we wrote this function to support the rasterio.mask() function whic requires a geojson object. So when you mask you will get a really weird output?? maybe a point. i dont even know. Could we raise a warning perhaps? OR should that be raised in crop_image() instead? maybe i'm thinking ahead of myself in terms of workflows and not seeing this as a basic general function! it's happened before.:)

Reply:
sure @mbjoseph do you mean in case they mix up min and max (as opposed to x and y)? it would be tricky i think to test for x vs y  :) but perhaps i dont understand! Does shapely just fix the order? i mean we *could* maybe do that too? it's all risky business however with boundaries as they are just numbers. the bounds object is a bit more explicit.

Reply:
Hmmmmm - that's worth thinking about. I think if we can decide how we want to handle this case, then we can figure out whether we want to raise an error, warning, or not. What's the behavior that you would want if you were masking a raster with a point? 

For me I would maybe expect to get one pixel back, with everything else being masked out (or if `invert=True`, everything except for that pixel). I have not tested this with `rasterio.mask` to see what happens in this case.

Reply:
i haven't either. i just hadn't thought about this at all until you brought this up here! i think we can have the conversation when we get to` crop_image` however as you are right. this just takes a geometry and makes a dictionary / geojson like thing. so that is what it does and it doesn't have to ask questions relative to other applications (like `mask()` . i'm complicating life here with my questions.

Reply:
Sounds good! I'm glad we're exploring these edge cases, and when we get to testing `crop_image`, we should definitely keep this one in mind.